omnigibson/utils/transform_utils.py

```diff
@@ -996,15 +988,34 @@ def vecs2axisangle(vec0, vec1):
     Converts the angle from unnormalized 3D vectors @vec0 to @vec1 into an axis-angle representation of the angle
 
     Args:
-        vec0 (3-array): (x,y,z) 3D vector, possibly unnormalized
-        vec1 (3-array): (x,y,z) 3D vector, possibly unnormalized
+        vec0 (np.array): (..., 3) (x,y,z) 3D vector, possibly unnormalized
+        vec1 (np.array): (..., 3) (x,y,z) 3D vector, possibly unnormalized
     """
     # Normalize vectors
-    vec0 = normalize(vec0)
-    vec1 = normalize(vec1)
+    vec0 = normalize(vec0, axis=-1)
+    vec1 = normalize(vec1, axis=-1)
+
+    # Get cross product for direction of angle, and multiply by arcos of the dot product which is the angle
+    return np.cross(vec0, vec1) * np.arccos((vec0 * vec1).sum(-1, keepdims=True))
+
+
+def vecs2quat(vec0, vec1, normalized=False):
+    """
+    Converts the angle from unnormalized 3D vectors @vec0 to @vec1 into a quaternion representation of the angle
+
+    Args:
+        vec0 (np.array): (..., 3) (x,y,z) 3D vector, possibly unnormalized
+        vec1 (np.array): (..., 3) (x,y,z) 3D vector, possibly unnormalized
+        normalized (bool): If True, @vec0 and @vec1 are assumed to already be normalized and we will skip the
+            normalization step (more efficient)
+    """
+    # Normalize vectors if requested
+    if not normalized:
+        vec0 = normalize(vec0, axis=-1)
+        vec1 = normalize(vec1, axis=-1)
 
     # Get cross product for direction of angle, and multiply by arcos of the dot product which is the angle
-    return np.cross(vec0, vec1) * np.arccos(np.dot(vec0, vec1))
+    return np.concatenate([np.cross(vec0, vec1), 1 + np.sum(vec0 * vec1, axis=-1, keepdims=True)], axis=-1)
```

Comment:
I am just rubber stamping on this and trusting you the math is correct lol.

Reply:
i'm going to revert to the original implementation lol

Reply:
taken from https://stackoverflow.com/questions/1171849/finding-quaternion-representing-the-rotation-from-one-vector-to-another!

also fixed a bug